src/setup-ocaml/cache.ts

```diff
@@ -94,10 +94,8 @@ async function composeOpamCacheKeys() {
 }
 
 function composeOpamDownloadCacheKeys() {
-  const repositories = OPAM_REPOSITORIES.map(([, u]) => {
-    const url = new URL(u);
-    const urn = path.join(url.hostname, url.pathname);
-    return urn;
+  const repositories = OPAM_REPOSITORIES.map(([n]) => {
```

Comment:
I don't think this will catch things correctly when the "value" (i.e. substance) changes.

Reply:
I wouldn't think the names are usually changed (and normally repositories is not changed). I think a fair expectation is that repos with different names would be separated and not share state, like cache.

If this doesn't convince you, would you prefer if I took the URL out of opam, or construct it in an ad-hoc way?

Reply:
I considered that, but I thought that the cache should always be as consistent as possible.